app/src/main/java/org/schabi/newpipe/database/feed/dao/FeedDAO.kt

```diff
@@ -66,6 +67,15 @@ abstract class FeedDAO {
             OR s.stream_type = 'LIVE_STREAM'
             OR s.stream_type = 'AUDIO_LIVE_STREAM'
         )
+        AND (
+            :includePartiallyPlayed
+            OR sh.stream_id IS NULL
+            OR sst.stream_id IS NULL
+            OR (sst.progress_time <= ${StreamStateEntity.PLAYBACK_SAVE_THRESHOLD_START_MILLISECONDS}
+            AND sst.progress_time <= s.duration * 1000 / 4)
+            OR (sst.progress_time >= s.duration * 1000 - ${StreamStateEntity.PLAYBACK_FINISHED_END_MILLISECONDS}
+            OR sst.progress_time >= s.duration * 1000 * 3 / 4)
```

Comment:
What you wanted to do here is the opposite of:
```
            OR sst.progress_time < s.duration * 1000 - ${StreamStateEntity.PLAYBACK_FINISHED_END_MILLISECONDS}
            OR sst.progress_time < s.duration * 1000 * 3 / 4
```
isn't it? That way you would get two disjoint sets to enable/disable via includePlayed and includePartiallyPlayed. Then, by De Morgan's law, it should be:
```suggestion
            OR (sst.progress_time >= s.duration * 1000 - ${StreamStateEntity.PLAYBACK_FINISHED_END_MILLISECONDS}
            AND sst.progress_time >= s.duration * 1000 * 3 / 4)
```

Also, small question, was it on purpose that the live stream checks were omitted? I mean, it makes sense, but just wanted to make sure

Reply:
Thanks for finding this flaw.
And yes, I intentionally omitted these checks.